### PR TITLE
Add comments to the resolv.conf file.

### PR DESCRIPTION
**Pull request by logicethos**
It can be confusing when you use the --dns x.x.x.x option on a container, to find it has not been added to resolv.conf

This is true for example, when you use the Macvlan network driver.  --dns is added to the Docker DNS server, and not resolv.conf.

I would suggest we see something like:

```
# --dns x.x.x.x has been added to Dockers internal DNS server
namserver 127.0.0.11
options ndots:0
```

or where no override setting made.

```
# File set by Docker for the internal DNS server
namserver 127.0.0.11
options ndots:0
```

### CODE REVIEW

**Comment by cpuguy83** [comment]
I agree this is confusing, however I'm not sure this is the right approach, particularly because this is just a library that could be consumed by anyone, as well as the `--dns` flag is specific to the docker CLI.

Maybe more generic like "if a custom dns setting was requested, it will be used by the internal DNS server" (or something to that affect).

Also note that your commits aren't signed, we can't accept a commit unless the commit is signed.
You can do so by simply doing `git commit -s --amend --no-edit` on your branch and re-push.

**Comment by logicethos** [comment]
This patch adds a comment field.  It's up to the calling code to set it or leave it empty.
Given golang doesn't support optional parameters, it's a breaking change.

A generic message would work, especially if it points to documentation.  It's a misunderstanding that  comes up a lot in the forums/issues.  AFAIK And there is no way to "inspect" the docker dns server from the command line to see what it's doing on a per-container bases.

File: resolvconf/resolvconf.go [hunk 1]
```diff
@@ -223,8 +223,20 @@ func GetOptions(resolvConf []byte) []string {
...
 	content := bytes.NewBuffer(nil)
+	if len(comments) > 0 {
+		commentLine := strings.Split(comments, "\n")
+		for i := range commentLine {	
+			if _, err := content.WriteString("# " + commentLine[i] + "\n"); err != nil {
+				return nil, err
+			}
+		}
+	} else {
+		if _, err := content.WriteString("# This file was created by Docker" + dns + "\n"); err != nil {
```
Comment (fcrisciani) [inline resolvconf/resolvconf.go]: the compilation fails for type missmatch here, dns is a []string so cannot be simply concatenated. you can try with strings.Join() eventually

File: resolvconf/resolvconf.go [hunk 2]
```diff
@@ -223,8 +223,20 @@ func GetOptions(resolvConf []byte) []string {
 // Build writes a configuration file to path containing a "nameserver" entry
 // for every element in dns, a "search" entry for every element in
 // dnsSearch, and an "options" entry for every element in dnsOptions.
-func Build(path string, dns, dnsSearch, dnsOptions []string) (*File, error) {
+func Build(path string, dns, dnsSearch, dnsOptions []string, comments string) (*File, error) {
```
Comment (fcrisciani) [inline resolvconf/resolvconf.go]: if there is multiple comments maybe []strings is more appropriate instead of doing the split on \n below

**Comment by corhere** [comment]
Will be carried by https://github.com/moby/moby/pull/47041

**Pull request closed without merging**